crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -37,6 +38,78 @@ impl DynamicScene {
         Self::from_world(&scene.world, type_registry)
     }
 
+    /// Create a Dynamic Scene with specific entities.
+    ///
+    /// The generic parameter is used as a `Query` filter.
+    ///
+    /// The created scene will include only the entities that match the query
+    /// filter provided. All components that impl `Reflect` will be included.
+    ///
+    /// # Example
+    ///
+    /// Here, the function is creating a scene with all the componentA's in the world, but none of the componentB's.
+    ///
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    /// use bevy_scene::DynamicScene;
+    ///
+    /// #[derive(Component)]
+    /// struct ComponentA;
+    ///
+    /// #[derive(Component)]
+    /// struct ComponentB;
+    ///
+    /// TODO how to add type registry to the world
+    /// let mut world = World::new();
+    ///
+    /// let my_scene = DynamicScene::from_query_filter::<(
+    ///   With<ComponentA>,
+    ///   Without<ComponentB>,
+    /// )>(&mut world);
+    /// ```
+    pub fn from_query_filter<F>(world: &mut World) -> Self
+    where
+        F: ReadOnlyWorldQuery + 'static,
+    {
+        let mut query = world.query::<Entity>();
```

Comment:
Should this be [`world.query_filtered::<Entity, F>()`](https://docs.rs/bevy/latest/bevy/prelude/struct.World.html#method.query_filtered)? I'm not sure where you're actually using the filter.

Reply:
Yeah thats a good point for sure. I tested it out and it works, but I am not sure I'm using the filter at all. Ill check it out.

Reply:
Yep, I want some aggressive test cases as part of this PR to ensure that this method does what it says.

Reply:
Yeah, it was wrong, I added a test case and fixed up the code!